### ISSUE TITLE
Formatting questions for pgHead

### ISSUE THREAD

**Issue by craigsapp** [music-encoding/music-encoding#496]
Here is an example `pgHead`:

```xml
<pgHead>
   <rend halign="center" valign="middle">
      <rend fontsize="x-large">Non è questa la mano (<rend fontstyle="italic">Rime</rend> 47)</rend>
   </rend>
   <rend halign="right" valign="bottom" fontsize="small">Ruggiero Giovannelli</rend>
</pgHead>
```

which renders in verovio like this:

<img width="903" alt="screen shot 2018-02-04 at 12 29 10 pm" src="https://user-images.githubusercontent.com/3487289/35782125-37a750ee-09a8-11e8-8174-c635c373947e.png">

----------------

The first question is that I want to add the composer's dates underneath the composer's name on the right side of the header:

<img width="1198" alt="screen shot 2018-02-04 at 12 32 37 pm" src="https://user-images.githubusercontent.com/3487289/35782136-566934e8-09a8-11e8-8ae0-268ac7fed0a5.png">

Doing this is obvious:

```xml
<pgHead>
   <rend halign="center" valign="middle">
      <rend fontsize="x-large">Non è questa la mano (<rend fontstyle="italic">Rime</rend> 47)</rend>
   </rend>
   <rend halign="right" valign="bottom" fontsize="small">Ruggiero Giovannelli</rend>
   <lb/>
   <rend halign="right" valign="bottom" fontsize="small"><rend fontstyle="italic">ca.</rend> 1560&#8211;1625</rend>
</pgHead>
```

But is it possible to allow centering the dates under the composer's name:

<img width="1198" alt="screen shot 2018-02-04 at 12 33 05 pm" src="https://user-images.githubusercontent.com/3487289/35782177-4ef87e16-09a9-11e8-8b5e-23cc206461df.png">

In other words create a group for the composer/dates which are mutually centered, and then right-justified as a group.

And how about this variant where the dates are left justified under the composer's name:

<img width="1194" alt="screen shot 2018-02-04 at 12 33 45 pm" src="https://user-images.githubusercontent.com/3487289/35782217-b2e426dc-09a9-11e8-8e3a-05349355ac9a.png">

which would be a grouping of the composer/dates left aligned to each other and then their grouping would be right-justified.

-------------------

A second question is: how to add vertical space between the title and the composer name:

<img width="1199" alt="screen shot 2018-02-04 at 12 34 31 pm" src="https://user-images.githubusercontent.com/3487289/35782234-ef0edbf2-09a9-11e8-81ee-0c2ad750a560.png">

**Comment by pe-ro** [comment]
I would expect the following markup to do what you're looking for --
```
<pgHead xmlns="http://www.music-encoding.org/ns/mei">
  <rend halign="center" valign="middle">
    <rend fontsize="x-large">Non è questa la mano (<rend fontstyle="italic">Rime</rend> 47)</rend>
  </rend>
  <lb/>
  <lb/>
  <lb/>
  <seg>
    <rend halign="right" valign="bottom" fontsize="small">
      Ruggiero Giovannelli<lb/>
      <rend halign="left">
        <rend fontstyle="italic">ca.</rend> 1560&#8211;1625
      </rend>    
    </rend>
  </seg>
</pgHead>
```
The nested `<rend halign="left">` for the dates should be left justified within the confines of the outer `<rend>`.  In other words, the "box" for the dates is left justified within its containing "box", which is right justified.

A nested `halign="center"` should work the same way --
```
<seg>
  <rend halign="right" valign="bottom" fontsize="small">
  Ruggiero Giovannelli<lb/>
    <rend halign="center">
      <rend fontstyle="italic">ca.</rend> 1560&#8211;1625
    </rend>
  </rend>
</seg>
```

Adding additional `<lb/>` elements between the title and the composer's name should result in more space.

**Comment by pe-ro** [comment]
It should work even without the containing `<seg>` --
```
<rend halign="right" valign="bottom" fontsize="small">
  Ruggiero Giovannelli<lb/>
  <rend halign="center">
    <rend fontstyle="italic">ca.</rend> 1560&#8211;1625
  </rend>
</rend>
```
Of course, it's up to Verovio to implement the markup assumptions.  😀

**Comment by lpugin** [comment]
> Of course, it's up to Verovio to implement the markup assumptions. 

Yes, this would make an excellent PR to receive 😀

**Comment by pe-ro** [comment]
It seems at this point at least one of the things that are missing in MEI's handling of this stuff is the ability to specify the *left margin* (and/or the position and size?) of a box that is itself right justified.  Perhaps adding `@position`, `@float`, `@clear`, `@visibility`, and `@z-index` to `<rend>` would be a way forward.

In any case, getting the following HTML to work
```
<!DOCTYPE html SYSTEM "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <title>Easy clearing method</title>
    <style type="text/css">
      #parent1 {
      border:5px solid purple;
      padding:10px;
      }
      .floatleft {
      border:5px solid red;
      float:left;
      width:200px;
      background:white;
      }
      .floatright {
      border:5px solid green;
      float:right;
      widht:200px;
      background:white;
      position:absolute;
      left:85%;
      }
      .clearfix:after{
      content: ".";
      display:block;
      height:0;
      clear:both;
      visibility:hidden;
      }
      .clearfix {display:inline-table;}
      /* Hide from IE-mac \*/
      * html .clearfix {height: 1%;}
      /* End hid
    </style>
  </head>
  <body>
    <h2 style="text-align:center">Non &egrave; questa la mano (<span style="font-style:italic">Rime</span> 47)</h2>
    <div id="parent1" class="clearfix">
      <span class="floatright">Ruggiero Giovannell.
        <span class="floatleft">ca. 1560&#8211;1625</span>
      </span>
    </div>
  </body>
</html>
```
as defined in my earlier comment would point out what we need to have in the MEI.  Someone who's more knowledgeable about CSS should take the lead on that, though.

Once we understand how to do it in HTML, would it be possible to "borrow" code/algorithms from one or more HTML browsers?

**Comment by pe-ro** [comment]
Another approach is to use `<table>` inside `<pgHead>` for layout.

**Comment by pe-ro** [comment]
Closing this for now, reopen if necessary.
